test/Graphics/StencilMode.test.cpp

```diff
@@ -0,0 +1,278 @@
+{
+    SECTION("Type traits")
+    {
+        STATIC_CHECK(std::is_copy_constructible_v<sf::StencilMode>);
+        STATIC_CHECK(std::is_copy_assignable_v<sf::StencilMode>);
+        STATIC_CHECK(std::is_nothrow_move_constructible_v<sf::StencilMode>);
+        STATIC_CHECK(std::is_nothrow_move_assignable_v<sf::StencilMode>);
+    }
+
+    SECTION("Construction")
+    {
+        SECTION("Default constructor")
+        {
+            const sf::StencilMode stencilMode;
+            CHECK(stencilMode.stencilComparison == sf::StencilComparison::Always);
+            CHECK(stencilMode.stencilUpdateOperation == sf::StencilUpdateOperation::Keep);
+            CHECK(stencilMode.stencilReference == 0);
+            CHECK(stencilMode.stencilMask == ~0u);
+            CHECK(stencilMode.stencilOnly == false);
+        }
+
+        SECTION("Verbose constructor")
+        {
+            const sf::StencilMode stencilMode(sf::StencilComparison::Equal, sf::StencilUpdateOperation::Replace, 1, 0u, true);
+            CHECK(stencilMode.stencilComparison == sf::StencilComparison::Equal);
+            CHECK(stencilMode.stencilUpdateOperation == sf::StencilUpdateOperation::Replace);
+            CHECK(stencilMode.stencilReference == 1);
+            CHECK(stencilMode.stencilMask == 0u);
+            CHECK(stencilMode.stencilOnly == true);
+        }
+    }
+
+    SECTION("Operators")
+    {
+        SECTION("operator==")
+        {
+            CHECK(sf::StencilMode() == sf::StencilMode());
+            CHECK(sf::StencilMode(sf::StencilComparison::Equal, sf::StencilUpdateOperation::Replace, 1, 0u, true) ==
+                  sf::StencilMode(sf::StencilComparison::Equal, sf::StencilUpdateOperation::Replace, 1, 0u, true));
+
+            CHECK_FALSE(sf::StencilMode() ==
+                        sf::StencilMode(sf::StencilComparison::Equal, sf::StencilUpdateOperation::Replace, 1, 0u, true));
+            CHECK_FALSE(sf::StencilMode(sf::StencilComparison::Greater, sf::StencilUpdateOperation::Invert, 0, ~0u, false) ==
+                        sf::StencilMode(sf::StencilComparison::Equal, sf::StencilUpdateOperation::Replace, 1, 0u, true));
+        }
+
+        SECTION("operator!=")
+        {
+            CHECK_FALSE(sf::StencilMode() != sf::StencilMode());
+            CHECK_FALSE(sf::StencilMode(sf::StencilComparison::Equal, sf::StencilUpdateOperation::Replace, 1, 0u, true) !=
+                        sf::StencilMode(sf::StencilComparison::Equal, sf::StencilUpdateOperation::Replace, 1, 0u, true));
+
+            CHECK(sf::StencilMode() !=
+                  sf::StencilMode(sf::StencilComparison::Equal, sf::StencilUpdateOperation::Replace, 1, 0u, true));
+            CHECK(sf::StencilMode(sf::StencilComparison::Greater, sf::StencilUpdateOperation::Invert, 0, ~0u, false) !=
+                  sf::StencilMode(sf::StencilComparison::Equal, sf::StencilUpdateOperation::Replace, 1, 0u, true));
+        }
+    }
+
+    sf::RenderTexture renderTexture;
```

Comment:
`StencilMode` itself is dead simple. It's just a simple struct. We just need to make sure the one constructor works and if we remove that constructor then we have even less code to test. I think that's where the tests should end.

If we want to test how it's used in rendering then I think those tests belong with the render texture tests.